mobile/src/components/setting/setting.tsx

```diff
@@ -38,20 +44,53 @@ import { styles } from './styles';
 
 const Settings: FC = () => {
   const dispatch = useAppDispatch();
-  const { userDetails, dataStatus } = useAppSelector(
-    (state) => state.userDetails,
-  );
+  const [selectedImage, setSelectedImage] = useState<Asset[] | null>();
+
+  const { user, userDataStatus, userDetails, userDetailsDataStatus } =
+    useAppSelector(({ auth, userDetails }) => ({
+      user: auth.user,
+      userDataStatus: auth.dataStatus,
+      userDetails: userDetails.userDetails,
+      userDetailsDataStatus: userDetails.dataStatus,
+    }));
 
   const maxDate = subtractYears(new Date(), UserAge.MIN);
   const minDate = subtractYears(new Date(), UserAge.MAX);
+  const selectionLimit = 1;
 
   const { control, errors, handleSubmit, reset } =
     useAppForm<UserDetailsUpdateInfoRequestDto>({
       defaultValues: DEFAULT_UPDATE_USER_DETAILS_PAYLOAD,
       validationSchema: userDetailsUpdateInfoValidationSchema,
     });
 
-  const handleCancel = (): void => reset();
+  const handleChooseAvatar = async (): Promise<void> => {
+    const image = await pickImage(selectionLimit);
```

Comment:
To select first element
```suggestion
    const [image] = await pickImage(selectionLimit) ?? [];
```